### ISSUE TITLE
training from scratch on own dataset

### ISSUE THREAD

**Issue by gdineshk6174** [CODEJIN/WaveNet#1]
Hello, my name is Dinesh, i am trying to build a TTS model using tacotron2(tacotron + wavenet) on my own dataset , i have reimplemented https://github.com/Rayhane-mamah/Tacotron-2 . the 1st model (tacotron) worked fine and the predicted spectrogram are good but the wavenet is overfitting . i have tried hyperparameter tuning on learning-rate and dropout rate but still of no use.i couldn't find any help in that repo's issues . <br>
i was thinking to train wavenet separately and since you have implemented wavenet in tensorflow2.0 and new , i would like to use your implementation of the model. so it would be really great if you could help me in providing the steps on how to train the your model from scratch on personal datasets.<br>
thank you

**Comment by CODEJIN** [comment]
Hi,

After seeing your issue, I added [a short paragraph](https://github.com/CODEJIN/WaveNet#if-you-want-to-generate-your-own-dataset) to the README.md about generating patterns. Of course, I think it is not sufficient, but I think it would be helpful to refer to it with the functions in [Pattern_Generate.py](https://github.com/CODEJIN/WaveNet/blob/master/Pattern_Generator.py). If you have any further questions about generating your own dataset, please leave a comment here again!

Thanks

Heejo

**Comment by gdineshk6174** [comment]
Hello Heejo, thank you the reply. im actually new to this so i would like to apologize in advance if i ask any noob questions. <br>so if im not wrong "Pattern_Generate.py " is like a preprocessing step which converts audio files(.wav) to spectrograms which would be input for the wavenet model and when we run "model.py" the wavenet starts training. am i right? and if i'm completely wrong , please kindly elaborate. 
<br> Thank you 
<br> Dinesh

**Comment by CODEJIN** [comment]
@gdineshk6174 Yes, you are right! `Pattern_Generate.py` makes pickled pattern files from wav files.
